### PR TITLE
feat(statsd source): add option to disable key sanitization

### PR DESCRIPTION
**Pull request by to-m**
By default the statsd source sanitizes metrics keys by removing/replacing certain chars. 
This is not the default behavior for fluentbit statsd source.

see: #20595  
This PR adds an option to disable key sanitization, leaving the default sanitize=true so as not to break backwards compatibility.

### CODE REVIEW

File: website/cue/reference/components/sources/base/statsd.cue [hunk 1]
```diff
@@ -59,6 +59,19 @@ base: components: sources: statsd: configuration: {
...
 		type: uint: unit: "bytes"
 	}
+	sanitize: {
+		description: """
+			sanitize
+			\tBy default all stat names (keys) are sanitized.
+			\t  "/" is replaced with "-"
+			\t  All whitespace is replaced with "_"
+			\t  All non alphanumeric characters [^a-zA-Z_\\-0-9\\.] are removed.
+			\tFor backwards compatibility this is remains the default behavior.
```
Comment (rtrieu) [inline website/cue/reference/components/sources/base/statsd.cue:69]:
```suggestion
			\tFor backwards compatibility this remains the default behavior.
```

**Review by rtrieu — APPROVED** [review]
Minor feedback but otherwise looks good.

File: src/sources/statsd/mod.rs [hunk 2]
```diff
@@ -113,6 +119,17 @@ pub struct TcpConfig {
...
     #[configurable(metadata(docs::type_unit = "connections"))]
     connection_limit: Option<u32>,
+
+    /// sanitize
+    ///	By default all stat names (keys) are sanitized.
+    ///	  "/" is replaced with "-"
+    ///	  All whitespace is replaced with "_"
+    ///	  All non alphanumeric characters [^a-zA-Z_\-0-9\.] are removed.
+    ///	For backwards compatibility this remains the default behavior.
+    ///	To disable sanitization, set to false
```
Comment (jszwedko) [inline src/sources/statsd/mod.rs:129]:
```suggestion
    ///	Whether to sanitize incoming statsd key names. When `true`, keys are sanitized by:
    /// - "/" is replaced with "-"
    /// - All whitespace is replaced with "_"
    ///- All non alphanumeric characters [^a-zA-Z_\-0-9\.] are removed.
```

This doc comment ends up in the generated documentation so I cleaned this up a bit. The default will be shown in the docs, so it doesn't need to be part of the comment.

**Review by jszwedko — COMMENTED** [review]
Nice, thanks @kates !

I left a suggestion in-line. Otherwise I think this just needs a changelog fragment added. See https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md

**Review by jszwedko — APPROVED** [review]
Thanks @to-m !

File: changelog.d/20595-statsd_source_key_sanitization.enhancement.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+The statsd source sanitizes stats keys, replacing '/' with '-', whitespace with '_' and removing all
+other non alphanumeric characters.
+
+The enhancement allows disabling this behavior.
```
Comment (jszwedko) [inline changelog.d/20595-statsd_source_key_sanitization.enhancement.md:4]:
```suggestion
The `statsd` source now has a configuration option to disable key sanitization: `sanitize`. The default is `true` to maintain backwards compatibility.

authors: to-m
```

**Review by jszwedko — COMMENTED** [review]
Actually, I left one suggestion for the changelog. Otherwise looks good!

**Pull request merged**